### DISCUSSION TITLE
Linker error in Windows implmentation

### DISCUSSION THREAD

**Discussion by escamoteur** [escamoteur/web_gpu#18] [General]
Hi @jpnurmi,

I tried to get the initialization going, but get Linker error the moment I try to call 

instance->GetAdapters()

could you have a look at this?

Thanks
Thomas

**Comment by jpnurmi** [comment]
Hi, where did you add the code, and what is "instance" referring to? Could you show git diff or something so I can see the context?

**Comment by jpnurmi** [comment]
GetAdapters symbol in  libdawn_native.dll vs. in the linker error message:
```
?GetAdapters@Instance@dawn_native@@QEBA?AV?$vector@VAdapter@dawn_native@@V?$allocator@VAdapter@dawn_native@@@__1@std@@@__1@std@@XZ

vs.

?GetAdapters@Instance@dawn_native@@QEBA?AV?$vector@VAdapter@dawn_native@@V?$allocator@VAdapter@dawn_native@@@std@@@std@@XZ
```

Both demangled:
```
public: class std::__1::vector<class dawn_native::Adapter,class std::__1::allocator<class dawn_native::Adapter> > __cdecl dawn_native::Instance::GetAdapters(void)const __ptr64

vs.

public: class std::vector<class dawn_native::Adapter,class std::allocator<class dawn_native::Adapter> > __cdecl dawn_native::Instance::GetAdapters(void)const __ptr64
```

==> `std::__1::vector` vs. `std::vector`?

**Comment by jpnurmi** [comment]
I think Clang libc++ vs. MSVC libstdc++

**Comment by jpnurmi** [comment]
I found a GN build config called `is_clang` for Dawn. Disabling it seems to do the trick, because then it builds the libs with MSVC. I have pushed Dawn libs built with MSVC: https://github.com/escamoteur/web_gpu/pull/19

**Reply by escamoteur** [reply]
You are a wizzard
